.github/workflows/verify_docker_image.yml

```diff
@@ -0,0 +1,40 @@
+name: Pull and Verify Docker Image
+
+on:
+  push:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  pull-and-build-image:
+    name: "Verify Docker Image"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Pull Docker Image
+        run: |
+          docker pull ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_gfx1201:main
+      - name: Run Docker Container and Execute Commands
+        run: |
+          docker run --rm ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_gfx1201:main /bin/bash -c "
+            echo 'Running inside the container';
+            echo 'Check Python version';
+            python3 --version;
+            echo 'List the pip packages';
+            pip list;
+            cd
+            echo 'Install wget';
+            apt-get install -y wget
+            echo 'Download smoke tests';
+            wget https://gist.githubusercontent.com/erman-gurses/ecab96a3f331ba6155b0eb17bbd2d84e/raw/3828aa1c6260b9bcac65f73c6bb5d193528c9eac/pytest-examples.py
```

Comment:
This is only running CPU tests. To test our docker image we should be using a GPU (device == CUDA?)

```python
import torch

def test_matrix_multiplication():
    matrix1 = torch.tensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]])
    matrix2 = torch.tensor([[7.0, 8.0, 9.0, 10.0], [11.0, 12.0, 13.0, 14.0], [15.0, 16.0, 17.0, 18.0]])
    expected = torch.tensor([[74.0, 80.0, 86.0, 92.0], [173.0, 188.0, 203.0, 218.0]])
    result = torch.mm(matrix1, matrix2)
    assert torch.allclose(result, expected)
```